doc/en/explanation/pythonpath.rst

```diff
@@ -38,26 +41,56 @@ these values:
   with ``prepend`` they would pick up the local version. This kind of confusion is why
-  we advocate for using :ref:`src <src-layout>` layouts.
+  we advocate for using :ref:`src-layouts <src-layout>`.
 
   Same as ``prepend``, requires test module names to be unique when the test directory tree is
   not arranged in packages, because the modules will put in :py:data:`sys.modules` after importing.
 
-* ``importlib``: new in pytest-6.0, this mode uses more fine control mechanisms provided by :mod:`importlib` to import test modules. This gives full control over the import process, and doesn't require changing :py:data:`sys.path`.
+* ``importlib``: this mode uses more fine control mechanisms provided by :mod:`importlib` to import test modules, without changing :py:data:`sys.path`.
 
-  For this reason this doesn't require test module names to be unique.
+  It works like this:
 
-  One drawback however is that test modules are non-importable by each other. Also,  utility
-  modules in the tests directories are not automatically importable because the tests directory is no longer
-  added to :py:data:`sys.path`.
+  1. Given a certain module path, for example ``tests/core/test_models.py``, derives a canonical name
+     like ``tests.core.test_models`` and tries to import it.
 
-  Initially we intended to make ``importlib`` the default in future releases, however it is clear now that
-  it has its own set of drawbacks so the default will remain ``prepend`` for the foreseeable future.
+     For test modules this will usually fail, as test modules are not usually importable, but will work for modules that are installed into a virtualenv.
+
+     For non-test modules this will usually work if they are installed/accessible via ``sys.path``, so
+     for example ``.env/lib/site-packages/app/core.py`` will be importable as ``app.core``. This is desirable
+     in general, and happens when plugins import non-test modules for other purposes, for example doctesting.
+
+     If this step succeeds, the module is returned.
+
+  2. If importing the module fails, we import the module directly using ``importlib``. Python requires the module
+     to also be available in ``sys.modules``, so pytest uses the derived unique name for it (``tests.core.test_models``)
+     and adds the module to ``sys.modules``.
+
+  Advantages of this mode:
+
+  * pytest will not change ``sys.path`` at all.
+  * Test module names do not need to be unique.
+
+  Drawbacks:
+
+  * that test modules are non-importable by each other.
+  * Test utility modules in the tests directories are not automatically importable because the tests directory is no longer
+    added to :py:data:`sys.path`. Note that by "test utility modules" we mean functions/classes which are imported by
+    other tests directly; this does not include fixtures, which should be placed in ``conftest.py`` files and are
+    discovered automatically by pytest.
```

Comment:
I don’t know if these are drawbacks. I think they encourage a pattern of separating importable test utilities and fixtures, which I think is a good thing. Pytest has its own namespace semantics, so mixing `conftest` and importable modules is confusing IMHO.

I think instead of labelling these as definite drawbacks, we should maybe add a section to the best practice page of how to deal with test utilities. E.g. I like to do this:

```
+ src/
|  + mymodule/*
|  + testing/mymodule/
|     + __init__.py  # public test utils, if any
|     + _private.py  # private test utils
+ test/*
```

But this if out of scope for this PR. If you’re interested, I cold try my hand at this.

Reply:
Indeed I had the same thought and was thinking about adding such a session later, feel free to tackle this if you want. 👍